gc/gc-tool-inttest/src/test/java/org/projectnessie/gc/tool/inttest/ITSparkIcebergNessieCLI.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.gc.tool.inttest;
+
+import static java.lang.String.format;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.CatalogProperties;
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.api.junit.jupiter.InjectSoftAssertions;
+import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.projectnessie.gc.tool.cli.util.RunCLI;
+import org.projectnessie.minio.Minio;
+import org.projectnessie.minio.MinioAccess;
+import org.projectnessie.minio.MinioExtension;
+import org.projectnessie.spark.extensions.SparkSqlTestBase;
+
+/** Full Nessie GC round trip test with Nessie, Spark, Iceberg, Minio. */
+@ExtendWith({MinioExtension.class, SoftAssertionsExtension.class})
+public class ITSparkIcebergNessieCLI extends SparkSqlTestBase {
+
+  public static final String JDBC_URL = "jdbc:h2:mem:gc_int_test;DB_CLOSE_DELAY=-1";
+
+  @InjectSoftAssertions private SoftAssertions soft;
+
+  @Minio static MinioAccess minio;
+
+  @TempDir Path tempDir;
+
+  @Override
+  protected String warehouseURI() {
+    return tempDir.toUri().toString();
+  }
+
+  @Override
+  protected Map<String, String> sparkHadoop() {
+    Map<String, String> r = new HashMap<>();
+    r.put("fs.s3.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem");
+    r.put("fs.s3a.access.key", minio.accessKey());
+    r.put("fs.s3a.secret.key", minio.secretKey());
+    r.put("fs.s3a.endpoint", minio.s3endpoint());
+    return r;
+  }
+
+  @Override
+  protected Map<String, String> nessieParams() {
+    Map<String, String> r = new HashMap<>(super.nessieParams());
+    r.put(CatalogProperties.FILE_IO_IMPL, "org.apache.iceberg.aws.s3.S3FileIO");
+    r.putAll(minio.icebergProperties());
+
+    System.setProperty("aws.region", "us-east-1");
+    System.setProperty("aws.s3.endpoint", minio.s3endpoint());
+    System.setProperty("aws.s3.accessKey", minio.accessKey());
+    System.setProperty("aws.s3.secretAccessKey", minio.secretKey());
+
+    return r;
+  }
+
+  @Test
+  public void fullRoundTrip(@TempDir Path workDir) throws Exception {
```

Comment:
I found an issue in this scenario

Have three commits, execute gc with defer-deletes and cutoff policy with retain 1 commit. Then delete the deferred deletes. Then execute gc again without defer option and it throws `FileNotFoundException`. It is probably Identifying the expired contents again from Nessie (this problem not happening for in-memory!)

Detailed steps:

create table nessie.db1.t1(id int) using iceberg;
insert into nessie.db1.t1 select 42;
insert into nessie.db1.t1 select 42;

./nessie-gc gc --defer-deletes --cutoff main=1 --jdbc-url jdbc:postgresql://127.0.0.1:5432/nessie_gc --jdbc-user pguser --jdbc-password mysecretpassword

./nessie-gc deferred-deletes -l 425c66b2-7e72-496d-b93d-07e96bd5dfc5  --jdbc-url jdbc:postgresql://127.0.0.1:5432/nessie_gc --jdbc-user pguser --jdbc-password mysecretpassword

./nessie-gc gc  --jdbc-url jdbc:postgresql://127.0.0.1:5432/nessie_gc --jdbc-user pguser --jdbc-password mysecretpassword


org.apache.iceberg.exceptions.NotFoundException: Failed to open input stream for file: /tmp/nessie2/db1/t1/metadata/00000-397fa47a-eb9a-477f-bcf3-4453dfde84fb.metadata.json
	at org.apache.iceberg.hadoop.HadoopInputFile.newStream(HadoopInputFile.java:184)
	at org.apache.iceberg.TableMetadataParser.read(TableMetadataParser.java:263)
	at org.apache.iceberg.TableMetadataParser.read(TableMetadataParser.java:258)
	at org.projectnessie.gc.iceberg.IcebergContentToFiles.extractFiles(IcebergContentToFiles.java:70)
	at org.projectnessie.gc.expire.PerContentDeleteExpired.lambda$identifyLiveFiles$2(PerContentDeleteExpired.java:125)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:271)
	at org.projectnessie.gc.contents.jdbc.JdbcHelper$ResultSetSplit.tryAdvance(JdbcHelper.java:115)
	at java.base/java.util.Spliterator.forEachRemaining(Spliterator.java:326)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.ReduceOps$5.evaluateSequential(ReduceOps.java:257)
	at java.base/java.util.stream.ReduceOps$5.evaluateSequential(ReduceOps.java:248)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.count(ReferencePipeline.java:605)
	at org.projectnessie.gc.expire.PerContentDeleteExpired.identifyLiveFiles(PerContentDeleteExpired.java:134)
	at org.projectnessie.gc.expire.PerContentDeleteExpired.expire(PerContentDeleteExpired.java:73)
	at org.projectnessie.gc.expire.local.DefaultLocalExpire.expireSingleContent(DefaultLocalExpire.java:104)

Reply:
I just narrowed it down into simple steps now and reproduced it with the in-memory itself.

```
create table nessie.db1.t1(id int) using iceberg;
insert into nessie.db1.t1 select 42;
insert into nessie.db1.t1 select 42;

./nessie-gc gc --cutoff main=1 --inmemory

./nessie-gc gc --cutoff --inmemory
```